### PR TITLE
Encryption and Rotation Updates

### PR DESCRIPTION
**Pull request by daxmc99**
Small changes to the secret package interface to allow for configuring key rotation and encryption

### CODE REVIEW

File: internal/secrets/encryptor.go [hunk 1]
```diff
@@ -69,43 +63,93 @@ func (e encryptor) EncryptBytes(plaintext []byte) (ciphertext []byte, err error)
...
+	plaintext, err := gcmDecrypt(ciphertext, e.primaryKey)
+	if err != nil { // perhaps it's already encrypted?
+		_, err = gcmDecrypt(ciphertext, e.secondaryKey)
+		if err == nil {
+			return ciphertext, nil
+		}
+		return ciphertext, err
+	}
+
+	return e.EncryptWithKey(plaintext, e.secondaryKey)
```
Comment (unknwon) [inline internal/secrets/encryptor.go:152]: Based on the docstring of the `primaryKey` and `secondaryKey`, the latter is the one we're deprecating, so we probably want to rotate in the opposite way as it is now?

Comment (unknwon) [reply internal/secrets/encryptor.go:152]: Also should add some tests for this new method.

File: internal/secrets/encryptor.go [hunk 2]
```diff
@@ -69,43 +69,94 @@ func (e encryptor) EncryptBytes(plaintext []byte) (ciphertext []byte, err error)
...
+func (e encryptor) RotateEncryption(ciphertext []byte) ([]byte, error) {
+	if !e.ConfiguredToRotate() {
+		return nil, &EncryptionError{errors.New("key rotation not configured")}
+	}
+	// try previous key first
+	plaintext, err := gcmDecrypt(ciphertext, e.secondaryKey)
+	if err != nil {
+		_, err = gcmDecrypt(ciphertext, e.primaryKey)
+		if err != nil {
+			return ciphertext, err
```
Comment (unknwon) [inline internal/secrets/encryptor.go:154]: nit: Unless there is a specific reason to return it, which should also need a code comment to explain why.
```suggestion
			return nil, err
```

File: internal/secrets/encryptor_test.go [hunk 3]
```diff
@@ -296,3 +298,62 @@ func Test_gatherKeys(t *testing.T) {
 		})
 	}
 }
+
+func Test_encryptor_RotateEncryption(t *testing.T) {
+
```
Comment (unknwon) [inline internal/secrets/encryptor_test.go:303]:
```suggestion
```

**Review by unknwon — APPROVED** [review]

File: internal/secrets/encryptor.go (hunk 2 above)
Comment (daxmc99) [reply internal/secrets/encryptor.go:154]: @chayim I believe there was a specific reason we wanted to return this?

Comment (chayim) [reply internal/secrets/encryptor.go:154]: No, the err check itself is valid.  Committed the suggestion and merging.

**Review by chayim — APPROVED** [review]

**Pull request merged**